### PR TITLE
JSDK-3094 custom logger definitions

### PR DESCRIPTION
**Pull request by PikaJoyce**
PR for custom logger type definition. 

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: tsdef/twilio-video-tests.ts [hunk 1]
```diff
@@ -1,5 +1,21 @@
 import * as Video from './index';
 
+function customLogging() {
+  const logger = Video.Logger.getLogger('twilio-video');
+
+  const originalFactory = logger.methodFactory;
+  logger.methodFactory = function(methodName, logLevel, loggerName) {
+    const method = originalFactory(methodName, logLevel, loggerName);
+
+    return function(datetime, logLevel, component, message, data) {
```
Comment (makarandp0) [inline tsdef/twilio-video-tests.ts:10]: NIT: datetime => dateTime

**Review by makarandp0 — APPROVED** [review]
just a small comment, but LGTM! I pulled your branch and saw VSCode auto-completing log methods!

File: tsdef/index.d.ts [hunk 2]
```diff
@@ -2,9 +2,11 @@ import { ConnectOptions, CreateLocalTrackOptions, CreateLocalTracksOptions, Loca
 import { LocalAudioTrack } from './LocalAudioTrack';
 import { LocalVideoTrack } from './LocalVideoTrack';
 import { Room } from './Room';
+import log from 'loglevel';
```
Comment (manjeshbhargav) [inline tsdef/index.d.ts:5]:
```suggestion
import Log from 'loglevel';
```

File: tsdef/index.d.ts [hunk 3]
```diff
@@ -2,9 +2,11 @@ import { ConnectOptions, CreateLocalTrackOptions, CreateLocalTracksOptions, Loca
...
 
 export const isSupported: boolean;
 export const version:string;
+export const Logger: typeof log;
```
Comment (manjeshbhargav) [inline tsdef/index.d.ts:9]:
```suggestion
export const Logger: typeof Log;
```

**Review by manjeshbhargav — APPROVED** [review]

File: tsdef/twilio-video-tests.ts [hunk 4]
```diff
@@ -1,5 +1,21 @@
...
 
+function customLogging() {
+  const logger = Video.Logger.getLogger('twilio-video');
+
+  const originalFactory = logger.methodFactory;
+  logger.methodFactory = function(methodName, logLevel, loggerName) {
+    const method = originalFactory(methodName, logLevel, loggerName);
+
+    return function(dateTime, logLevel, component, message, data) {
+            method(dateTime, logLevel, component, message, data);
```
Comment (manjeshbhargav) [inline tsdef/twilio-video-tests.ts:11]:
```suggestion
    method(dateTime, logLevel, component, message, data);
```

**Review by manjeshbhargav — APPROVED** [review]

File: tsdef/types.d.ts [hunk 5]
```diff
@@ -145,6 +145,7 @@ export interface MediaStreamTrackPublishOptions {
 /**
  * @param {dscpTagging}: Deprecated
  * @param {eventListener}: Deprecated
+ * @param {logLevels}: Deprecated
```
Comment (makarandp0) [inline tsdef/types.d.ts:148]:
```suggestion
 * @param {logLevel}: Deprecated
```

**Review by makarandp0 — CHANGES_REQUESTED** [review]
I missed this one earlier. I think the name of parameter is `logLevel`

**Review by makarandp0 — APPROVED** [review]
🏆

**Pull request merged**